### ISSUE TITLE
Feature Request: Add ability to take screenshots

### ISSUE THREAD

**Issue by Sebastian-Hirsch** [Rafostar/clapper#113]
### Description

Clapper has all the features i know and need of mpv besides two. The second feature I miss is the ability to take screenshots. This is also nicely implemented in mpv and is an feature I use really often. It should have some keyboard shortcuts to make a screenshot (s key in mpv). Options to change the screenshot path and the generated filename would also be great!

### Systeminfo

OS: Fedora 34
Package: Flatpak
GPU Driver: Intel Mesa Opensource, iHD VAAPI Media Driver

**Comment by Rafostar** [comment]
> Options to change the screenshot path and the generated filename would also be great!

This could be solved by introducing 2 shortcuts:
* `s` - quick save (autogen filename in configured folder)
* `Ctrl+s` - opens file chooser to select where and how name the file to save.

**Comment by gregthwuen** [comment]
I second this request – one of the main features I'm still missing :)

Regarding the shortcuts: Using `Ctrl+s` for choosing the location doesn't feel right to me. I would expect it to do something else, when `s` is enough for quick saving a screenshot, but maybe that's just me.

Maybe `Alt+s` would be a viable alternative? And `s` and `Shift+s` could replicate mpv's feature to save with or without subtitles embedded, if the implementation of the screenshot function doesn't make that to hard :)

For reference: In VLC, `Shift+s` is for taking a screenshot while `Ctrl+s` starts streaming the video in the network. In mpv, `Ctrl+s` is for taking a screenshot _including_ the OSD while `s` and `Shift+s` are for saving a screenshot with and without subtitles, respectively.

**Comment by x5f3759df** [comment]
I discover this player recently and its absolutely amazing, but I miss ability to take screenshots a lot. Will be fine to have this. Totem has its feature, but it awful to use and Clapper seems like much better alternative. Don't wanna use MPV because VAAPI performs better with GST on my machine.

**Comment by Rafostar** [comment]
> For reference: In VLC, Shift+s is for taking a screenshot while Ctrl+s starts streaming the video in the network. In mpv, Ctrl+s is for taking a screenshot including the OSD while s and Shift+s are for saving a screenshot with and without subtitles, respectively.

Clapper does not do streaming so we have Ctrl+s free. I guess best would be:
* s - quick screenshot
* Shift + s - quick without subtitles
* Ctrl + s - custom (with select where to save file chooser) with subtitles
* Ctrl + Shift + s - custom without subtitles

i.e. `Shift` controls if subtitles should be included, `Ctrl` - if open dialog to select save location

Something like this seems clear and quick to learn to me.

**Comment by gregthwuen** [comment]
Sounds good to me, only mpv has it the other way around (Shift + s for saving with subtitles), see: https://mpv.io/manual/master/. I guess I wasn't clear about that in my original comment.

Of course Clapper doesn't need to follow mpv's implementation, but it seems like for a lot of users Clapper is a direct replacement for mpv/Celluloid.

**Comment by Rafostar** [comment]
> Sounds good to me, only mpv has it the other way around (Shift + s for saving with subtitles), see: https://mpv.io/manual/master/#keyboard-control. I guess I wasn't clear about that in my original comment.

Good point. Its gonna be some work to have a split support like this for with/without subtitles, but I rather have this implemented once and good and not quick/bad. So not immediately, but we gonna decide which type of screenshot should be default one (with/without subtitles).

**Comment by Rafostar** [comment]
@gregthwuen 

> only mpv has it the other way around (Shift + s for saving with subtitles)

I took a glance at mpv key shortcuts and it says opposite:
```
By default, S takes screenshots without subtitles, while s includes subtitles
```

**Comment by gregthwuen** [comment]
Oh thanks for catching that, must've misread it like three times. Then I have no more complaints, and thank you for considering the feature!
